README.md

```diff
@@ -1,5 +1,189 @@
+    * `param`, the configuration parameter name;
+    * `defaultValue`.
+
+### Environment Configuration Parameter
+
+* `isProduction()`, returns true if the environment is production;
+* `isDevelopment()`, returns true if the environment is **not** production;
+* `isTest()`, returns true if the environment is test.
+
+----
+
+### Validators and Transformers
+
+Ad-hoc validators and transformers can be built leveraging the `EnvValueValidator` and the `EnvValueTransformer` types exposed by the library. Alternatively, the following validators and transformers are already provided out of the box:
+
+#### Validators
+
+* `createRangeValidator()`, which accepts `greaterOrEqualThan` and `lessOrEqualThan` and validates that a numeric value ranges between those numbers.
+
+#### Transformers
+
+* `ensureClosingSlashTransformer()`, which accepts a `value` as parameter, that can be a string or nullable, and adds a closing slash if it is missing and the value is defined.
+
+## Error Handling
+
+The library provides classes and methods for error handling.
+
+### Gloabl Error Handler
+
+Public methods to leverage a global error handler are provided to be used when the process is run outside of the context of the request (e. g. in a queue where no one would catch an error if thrown):
+* `resolveGlobalErrorLogObject()`, which accepts `err` and optionally `correlationID` as parameters and converts the plain error into a serializable object. If the error is not a built-in `Error` type and doesn't have any message, a fixed string is returned instead;
+* `executeAndHandleGlobalErrors()`, which accepts the `operation` parameter and will return the result of executing such operation. If an error is thrown during the execution of the operation, `resolveGlobalErrorLogObject()` is called to log the error and the process is terminated;
+* `executeAsyncAndHandleGlobalErrors()`, which accepts `operation` and optionally `stopOnError` as parameters and will return the result of executing such operation **asynchronously**. If an error is thrown during the execution of the operation, `resolveGlobalErrorLogObject()` is called to log the error and the process is terminated only if `stopOnError` is `true`. `stopOnError` defaults to `true` if not provided.
+
+### Errors
+
+The library exposes classes for the following errors:
+* `InternalError`, which issues a `500` status code and is not exposed in the global error handler. It expects the following parameters:
+    * `message`;
+    * `errorCode`;
+    * `details` – (optional).
+* `PublicNonRecoverableError`, which issues the HTTP status code provided and signals that the user did something wrong, hence the error is returned to the consumer of the API. It expects the following parameters:
+    * `message`;
+    * `errorCode`;
+    * `details` – (optional);
+    * `httpStatusCode` – (optional). Defaults to `500`;
+
+### Either
+
```

Comment:
adding link to an article with longer explanation of the pattern might be helpful (it is there in the code, I think)